util.hpp

```diff
@@ -0,0 +1,22 @@
+#include <cassert>
+#include <memory>
+#include <string>
+#include <vector>
+#include <chrono>
+#include <iostream>
+#include <algorithm>
+
+using timer = std::chrono::high_resolution_clock;
+using usecs = std::chrono::microseconds;
+
+static double delta_t(const timer::time_point t0, const timer::time_point t1) {
+  return std::chrono::duration_cast<usecs>(t1 - t0).count();
+}
+
+struct benchmark_parameters {
+  size_t threads, blocks;
+  size_t repetitions, epochs;
```

Comment:
What's the difference between repetitions and epochs?

Reply:
repetitions iterate over all epoch. We could -- alternatively -- time each epoch, but that would remove one level of fork/join in the graphs.

Reply:
I don't quite understand what you mean by epoch then. I thought the graph would execute completely within a single epoch? Then repetitions and epochs are just the same thing.

Reply:
Addressed offline.